### PR TITLE
Task/moko 1438 moko resources

### PR DESCRIPTION
**Pull request by DenielGrin**

### CODE REVIEW

File: shared/build.gradle.kts [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
 }
 
 version = "1.0-SNAPSHOT"
 
+repositories {
+    gradlePluginPortal()
+    google()
+    mavenCentral()
+}
+
```
Comment (Alex009) [inline shared/build.gradle.kts]: it's not needed here

File: build.gradle.kts [hunk 2]
```diff
@@ -9,6 +9,16 @@ plugins {
...
+buildscript {
+    repositories {
+        gradlePluginPortal()
+    }
+
+    dependencies {
+        classpath("dev.icerock.moko:resources-generator:0.20.1")
+    }
+}
+
```
Comment (Alex009) [inline build.gradle.kts]: let's use same way as up - with plugins block.
```
plugins {
  id("dev.icerock.mobile.multiplatform-resources") apply false
}
```

and in settings.gradle.kts add info about version

File: shared/build.gradle.kts [hunk 3]
```diff
@@ -26,7 +33,8 @@ kotlin {
             baseName = "shared"
             isStatic = true
         }
-        extraSpecAttributes["resources"] = "['src/commonMain/resources/**', 'src/iosMain/resources/**']"
+        extraSpecAttributes["resources"] =
+            "['src/commonMain/resources/**', 'src/iosMain/resources/**']"
```
Comment (Alex009) [inline shared/build.gradle.kts]: this part not needed with moko resources i think

File: shared/build.gradle.kts [hunk 4]
```diff
@@ -39,17 +47,22 @@ kotlin {
                 implementation(compose.components.resources)
```
Comment (Alex009) [inline shared/build.gradle.kts]: this lib not needed with moko resources i think

File: shared/build.gradle.kts [hunk 5]
```diff
@@ -39,17 +47,22 @@ kotlin {
...
             }
         }
+
         val androidMain by getting {
             dependencies {
                 api("androidx.activity:activity-compose:1.6.1")
                 api("androidx.appcompat:appcompat:1.6.1")
                 api("androidx.core:core-ktx:1.9.0")
+                api("dev.icerock.moko:resources-compose:0.20.1")
```
Comment (Alex009) [inline shared/build.gradle.kts]: in commonMain it should be

File: shared/build.gradle.kts [hunk 6]
```diff
@@ -58,6 +71,15 @@ kotlin {
     }
 }
 
+dependencies {
+    commonMainApi("dev.icerock.moko:resources:0.20.1")
```
Comment (Alex009) [inline shared/build.gradle.kts]: please use same way as you see for other deps

File: shared/src/androidMain/kotlin/main.android.kt [hunk 7]
```diff
@@ -1,5 +0,0 @@
-import androidx.compose.runtime.Composable
-
-actual fun getPlatformName(): String = "Android"
-
-@Composable fun MainView() = App()
```
Comment (Alex009) [inline shared/src/androidMain/kotlin/main.android.kt]: why it's deleted? rollback please

File: shared/src/androidMain/kotlin/org/icerock/template/main.android.kt [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+import dev.icerock.moko.resources.FontResource
+import dev.icerock.moko.resources.desc.desc
+import org.icerock.template.MR
+
+@SuppressLint("StaticFieldLeak")
+private var context: Context = AppCompatActivity()
+
+actual fun getMokoResourcesText(): String = MR.strings.moko_resources_title.desc().toString(context)
+actual fun getMokoResourcesColor(): Color = MR.colors.moko_color_title.color
+actual fun getMokoResourcesFont(): Int? = MR.fonts.Roboto.bold.getTypeface(context)?.style
```
Comment (Alex009) [inline shared/src/androidMain/kotlin/org/icerock/template/main.android.kt]: not needed at all.

File: shared/src/androidMain/kotlin/org/icerock/template/main.android.kt [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.LocalContext
+import dev.icerock.moko.graphics.Color
+import dev.icerock.moko.resources.FontResource
+import dev.icerock.moko.resources.desc.desc
+import org.icerock.template.MR
+
+@SuppressLint("StaticFieldLeak")
+private var context: Context = AppCompatActivity()
```
Comment (Alex009) [inline shared/src/androidMain/kotlin/org/icerock/template/main.android.kt]: never do this!

read - https://developer.android.com/jetpack/compose/compositionlocal

File: shared/src/commonMain/kotlin/App.kt [hunk 10]
```diff
@@ -6,18 +9,77 @@ import androidx.compose.runtime.getValue
...
+            }
         }
     }
 }
 
-expect fun getPlatformName(): String
\ No newline at end of file
+expect fun getMokoResourcesText(): String
+expect fun getMokoResourcesColor(): GraphicsColor
+expect fun getMokoResourcesFont(): Int?
```
Comment (Alex009) [inline shared/src/commonMain/kotlin/App.kt]: we should use functions from `resources-compose`

File: shared/src/desktopMain/kotlin/main.desktop.kt [hunk 11]
```diff
@@ -1,12 +0,0 @@
-import androidx.compose.desktop.ui.tooling.preview.Preview
```
Comment (Alex009) [inline shared/src/desktopMain/kotlin/main.desktop.kt]: revert this change please

File: shared/src/desktopMain/kotlin/org/icerock/template/main.desktop.kt [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import androidx.compose.runtime.Composable
+import dev.icerock.moko.graphics.Color
+import org.icerock.template.MR
+
+actual fun getMokoResourcesText(): String = MR.strings.moko_resources_title.localized()
+actual fun getMokoResourcesColor(): Color = MR.colors.moko_color_title.color
+actual fun getMokoResourcesFont(): Int? = MR.fonts.Roboto.bold.font.style
```
Comment (Alex009) [inline shared/src/desktopMain/kotlin/org/icerock/template/main.desktop.kt]: we should not write any actuals

File: shared/src/iosMain/kotlin/main.ios.kt [hunk 13]
```diff
@@ -1,5 +0,0 @@
-import androidx.compose.ui.window.ComposeUIViewController
```
Comment (Alex009) [inline shared/src/iosMain/kotlin/main.ios.kt]: revert this change please

File: shared/src/iosMain/kotlin/org/icerock/template/main.ios.kt [hunk 14]
```diff
@@ -0,0 +1,9 @@
+import androidx.compose.ui.window.ComposeUIViewController
+import dev.icerock.moko.graphics.Color
+import dev.icerock.moko.resources.desc.desc
+import org.icerock.template.MR
+
+actual fun getMokoResourcesText(): String = MR.strings.moko_resources_title.desc().localized()
+actual fun getMokoResourcesColor(): Color = MR.colors.moko_color_title.color
```
Comment (Alex009) [inline shared/src/iosMain/kotlin/org/icerock/template/main.ios.kt]: we should not use actual here

File: shared/build.gradle.kts [hunk 15]
```diff
@@ -58,6 +71,15 @@ kotlin {
...
+}
+
+multiplatformResources {
+    multiplatformResourcesPackage = "org.icerock.template"
```
Comment (Alex009) [inline shared/build.gradle.kts]: org.icerock? no.
`com.myapplication.common` - as we see in `AndroidManifest`

**Review by Alex009 — CHANGES_REQUESTED** [review]

File: shared/build.gradle.kts (hunk 1 above)
Comment (DenielGrin) [reply shared/build.gradle.kts]: done,deleted

File: build.gradle.kts (hunk 2 above)
Comment (DenielGrin) [reply build.gradle.kts]: agree, changed it

File: shared/build.gradle.kts (hunk 3 above)
Comment (DenielGrin) [reply shared/build.gradle.kts]: done, deleted

File: shared/build.gradle.kts (hunk 4 above)
Comment (DenielGrin) [reply shared/build.gradle.kts]: deleted

File: shared/build.gradle.kts (hunk 5 above)
Comment (DenielGrin) [reply shared/build.gradle.kts]: done, change it

File: shared/src/desktopMain/kotlin/main.desktop.kt (hunk 11 above)
Comment (DenielGrin) [reply shared/src/desktopMain/kotlin/main.desktop.kt]: done

File: shared/src/iosMain/kotlin/main.ios.kt (hunk 13 above)
Comment (DenielGrin) [reply shared/src/iosMain/kotlin/main.ios.kt]: done

File: shared/src/iosMain/kotlin/org/icerock/template/main.ios.kt (hunk 14 above)
Comment (DenielGrin) [reply shared/src/iosMain/kotlin/org/icerock/template/main.ios.kt]: deleted

File: shared/build.gradle.kts (hunk 15 above)
Comment (DenielGrin) [reply shared/build.gradle.kts]: done, changed namespace

File: androidApp/build.gradle.kts [hunk 16]
```diff
@@ -10,6 +10,8 @@ kotlin {
         val androidMain by getting {
             dependencies {
                 implementation(project(":shared"))
+                implementation("dev.icerock.moko:resources-compose:0.20.1")
```
Comment (Alex009) [inline androidApp/build.gradle.kts]: we not use this in android at now

File: shared/build.gradle.kts [hunk 17]
```diff
@@ -35,21 +35,25 @@ kotlin {
                 implementation(compose.runtime)
                 implementation(compose.foundation)
                 implementation(compose.material)
-                @OptIn(org.jetbrains.compose.ExperimentalComposeLibrary::class)
-                implementation(compose.components.resources)
+                api("dev.icerock.moko:resources:0.20.1")
```
Comment (Alex009) [inline shared/build.gradle.kts]: version should be from gradle.properties as we use same way that in original template

File: shared/src/androidMain/kotlin/org/icerock/template/main.android.kt (hunk 9 above)
Comment (Alex009) [reply shared/src/androidMain/kotlin/org/icerock/template/main.android.kt:10]: not changed :(

File: shared/src/commonMain/kotlin/App.kt [hunk 18]
```diff
@@ -6,18 +9,39 @@ import androidx.compose.runtime.getValue
...
+            ) {
+                Text("Get moko text")
+            }
         }
     }
 }
 
-expect fun getPlatformName(): String
\ No newline at end of file
+expect fun getMokoResourcesText(): String
```
Comment (Alex009) [inline shared/src/commonMain/kotlin/App.kt]: we should not use expect, all expects should be in moko-resources lib

**Review by Alex009 — CHANGES_REQUESTED** [review]

File: androidApp/build.gradle.kts (hunk 16 above)
Comment (DenielGrin) [reply androidApp/build.gradle.kts]: deleted

File: shared/src/commonMain/kotlin/App.kt (hunk 18 above)
Comment (DenielGrin) [reply shared/src/commonMain/kotlin/App.kt]: deleted, return previous name

File: shared/build.gradle.kts (hunk 17 above)
Comment (DenielGrin) [reply shared/build.gradle.kts]: agree, changed

**Comment by Alex009** [comment]
@BorisTrubitsyn please [update library to 0.21.0](https://github.com/icerockdev/moko-resources/releases/tag/release%2F0.21.0). and add usages of all resources, like here  - https://github.com/icerockdev/moko-resources/blob/master/samples/compose-resources-gallery/shared/src/commonMain/kotlin/App.kt

File: shared/build.gradle.kts [hunk 19]
```diff
@@ -26,30 +27,35 @@ kotlin {
...
+            val mokoResourcesVersion = extra["moko.resources.version"] as String
+
             dependencies {
                 implementation(compose.runtime)
                 implementation(compose.foundation)
                 implementation(compose.material)
-                @OptIn(org.jetbrains.compose.ExperimentalComposeLibrary::class)
-                implementation(compose.components.resources)
+                api("dev.icerock.moko:resources:${mokoResourcesVersion}")
+                api("dev.icerock.moko:resources-compose:0.21.0")
```
Comment (Alex009) [inline shared/build.gradle.kts]: version should be `mokoResourcesVersion`

File: shared/src/androidMain/kotlin/org/icerock/template/main.android.kt [hunk 20]
```diff
@@ -2,4 +2,5 @@ import androidx.compose.runtime.Composable
 
 actual fun getPlatformName(): String = "Android"
 
-@Composable fun MainView() = App()
+@Composable
+fun MainView() = App()
```
Comment (Alex009) [inline shared/src/androidMain/kotlin/org/icerock/template/main.android.kt]: rollback please this change. and moving of file too

File: shared/src/desktopMain/kotlin/org/icerock/template/main.desktop.kt [hunk 21]
```diff
@@ -3,10 +3,11 @@ import androidx.compose.runtime.Composable
 
 actual fun getPlatformName(): String = "Desktop"
 
-@Composable fun MainView() = App()
+@Composable
+fun MainView() = App()
```
Comment (Alex009) [inline shared/src/desktopMain/kotlin/org/icerock/template/main.desktop.kt]: rollback all changes of this file please

File: shared/src/iosMain/kotlin/org/icerock/template/main.ios.kt [hunk 22]
```diff
@@ -0,0 +1,5 @@
+import androidx.compose.ui.window.ComposeUIViewController
+
+actual fun getPlatformName(): String = "IOS"
+
+fun MainViewController() = ComposeUIViewController { App() }
```
Comment (Alex009) [inline shared/src/iosMain/kotlin/org/icerock/template/main.ios.kt]: rollback all changes of this file too

**Review by Alex009 — CHANGES_REQUESTED** [review]

File: shared/src/iosMain/kotlin/org/icerock/template/main.ios.kt (hunk 22 above)
Comment (DenielGrin) [reply shared/src/iosMain/kotlin/org/icerock/template/main.ios.kt]: done

File: shared/src/desktopMain/kotlin/org/icerock/template/main.desktop.kt (hunk 21 above)
Comment (DenielGrin) [reply shared/src/desktopMain/kotlin/org/icerock/template/main.desktop.kt]: done

File: shared/src/androidMain/kotlin/org/icerock/template/main.android.kt (hunk 20 above)
Comment (DenielGrin) [reply shared/src/androidMain/kotlin/org/icerock/template/main.android.kt]: done

File: shared/build.gradle.kts (hunk 19 above)
Comment (DenielGrin) [reply shared/build.gradle.kts]: fixed

**Comment by Alex009** [comment]
@BorisTrubitsyn need to fix conflicts

**Review by Alex009 — APPROVED** [review]
